### PR TITLE
Made manifest contract_id work with both namespaced and non-namespaced contract IDs

### LINKED ISSUE #1137: Make manifest `contract_id` work with both namespaced and non-namespaced contract IDs
- Namespaced contract ID: `fuels....`
- Not namespaced contract ID: `0x0000...`
- The `contract_id` field in the indexer manifest should work with both 
- We should also update the default indexer manifest with an additional comment for the `contract_id` field specifying that we support both contract ID types
  - That is [here](https://github.com/FuelLabs/fuel-indexer/blob/develop/plugins/forc-index/src/defaults.rs#L111)
- If we implement this, we can close https://github.com/FuelLabs/fuel-indexer/issues/983

### PR DESCRIPTION
**Pull request by dadepo**
Closes [#1137](https://github.com/FuelLabs/fuel-indexer/issues/1137)

### Description

Made manifest contract_id work with both namespaced and non-namespaced contract IDs

### Changelog

Updated the logic that parses the address in the manifest to Bech32ContractId to work with both namespace and non-space addresses.

### CODE REVIEW

File: examples/fuel-explorer/fuel-explorer/fuel_explorer.manifest.yaml [hunk 1]
```diff
@@ -22,6 +22,8 @@ end_block: ~
 fuel_client: ~
 
 # The contract_id specifies which particular contract you would like your index to subscribe to.
+# Both namespaced contract IDs (e.g., 'fuels...') and
```
Comment (ra0x3) [inline examples/fuel-explorer/fuel-explorer/fuel_explorer.manifest.yaml]: We can remove this added comment

File: examples/hello-world-native/hello-indexer-native/hello_indexer_native.manifest.yaml [hunk 2]
```diff
@@ -22,6 +22,8 @@ end_block: ~
 fuel_client: ~
 
 # The contract_id specifies which particular contract you would like your index to subscribe to.
+# Both namespaced contract IDs (e.g., 'fuels...') and
```
Comment (ra0x3) [inline examples/hello-world-native/hello-indexer-native/hello_indexer_native.manifest.yaml]: We can remove this added comment

File: examples/hello-world/hello-indexer/hello_indexer.manifest.yaml [hunk 3]
```diff
@@ -22,6 +22,8 @@ end_block: ~
 fuel_client: ~
 
 # The contract_id specifies which particular contract you would like your index to subscribe to.
+# Both namespaced contract IDs (e.g., 'fuels...') and
```
Comment (ra0x3) [inline examples/hello-world/hello-indexer/hello_indexer.manifest.yaml]: We can remove this added comment

File: packages/fuel-indexer-macros/src/indexer.rs [hunk 4]
```diff
@@ -255,8 +255,10 @@ fn process_fn_items(
                 .iter()
                 .map(|id| {
                     quote! {
-                        let id_bytes = <[u8; 32]>::try_from(#id).expect("Could not convert contract ID into bytes");
-                        Bech32ContractId::new("fuel", id_bytes)
+                        Bech32ContractId::from_str(#id).unwrap_or_else(|_| {
```
Comment (ra0x3) [inline packages/fuel-indexer-macros/src/indexer.rs:397]: 👌🏽

File: packages/fuel-indexer-tests/indexers/fuel-indexer-test/fuel_indexer_test.yaml [hunk 5]
```diff
@@ -4,7 +4,9 @@ graphql_schema: packages/fuel-indexer-tests/indexers/fuel-indexer-test/schema/fu
 abi: packages/fuel-indexer-tests/contracts/fuel-indexer-test/out/debug/fuel-indexer-test-abi.json
 start_block: ~
 end_block: ~
-contract_id: fuel12zfku575m48qgctuca7lf5rxwnrv93nn3yfxxtdte9hqqj7v45vsuujmpt
+contract_id:
+  - fuel1p4g0suykga85824l4x465haxtg7n6rlefxv2tch8av5g6ry88pxszp6aj0
+  - 0x0d50f87096474f43aabfa9abaa5fa65a3d3d0ff94998a5e2e7eb288d0c87384d
```
Comment (ra0x3) [inline packages/fuel-indexer-tests/indexers/fuel-indexer-test/fuel_indexer_test.yaml]: Let's remove `fuel1...` and just use `0x0d5...` - and if tests pass, that proves this works :)

File: plugins/forc-index/src/defaults.rs [hunk 6]
```diff
@@ -112,6 +112,8 @@ end_block: ~
 fuel_client: ~
 
 # The contract_id specifies which particular contract you would like your index to subscribe to.
+# Both namespaced contract IDs (e.g., 'fuels...') and
```
Comment (ra0x3) [inline plugins/forc-index/src/defaults.rs]: We can remove this as well

**Review by ra0x3 — CHANGES_REQUESTED** [review]
@dadepo 
- Looks mostly ok, left some minor feedback
- If you want to add a blurb to the docs, you can open `manifest.md`, go to the `contract_id` section, an add the following as a quote/blurb:

> Note that this parameter supports both Bech32 contract IDs and non-Bech32 contract IDs

**Comment by ra0x3** [comment]
@dadepo 

- This looks ok. Give us a bit of time and we'll get it merged for you :) 
- Really need to spend some time to figure out how to get 3rd-party contributors to run CI using the same tokens we (internal contributors) use

**Review by deekerno — APPROVED** [review]
I manually tested this and it works. 👌🏽

**Pull request merged**